remote_engine_client/src/client.rs

```diff
@@ -239,6 +247,114 @@ impl Client {
         Ok(results)
     }
 
+    pub async fn alter_table_schema(&self, request: AlterTableSchemaRequest) -> Result<()> {
+        // Find the channel from router firstly.
+        let route_context = self.cached_router.route(&request.table_ident).await?;
+
+        let table_ident = request.table_ident.clone();
+        let request_pb: ceresdbproto::remote_engine::AlterTableSchemaRequest = request.into();
+        let mut rpc_client = RemoteEngineServiceClient::<Channel>::new(route_context.channel);
+
+        let mut result = Ok(());
+        // Alter schema to remote engine with retry.
+        for i in 0..(self.max_retry + 1) {
```

Comment:
I don't see this retry merchanism in other implementations, but the `alter_table_options` shares the duplicate codes. So shall we make it(the retyr logic) as a macro or a method to avoid duplicate codes.

Reply:
`alter table` can be idempotent, so retry logic is added. Other operations are not idempotent.

Reply:
Add TODO.